src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0 && separator !== undefined) {
+        result += separator;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
```

Comment:
You can simplify your code by using a ternary operator. Instead of using if/else, consider writing 'result += this[i] === null || this[i] === undefined ? '' : this[i]'.